features.tt

```diff
@@ -47,26 +47,25 @@
       <h2>Complete dependencies</h2>
 
       <p>Nix helps you make sure that package dependency specifications are
-        complete. In general, when you’re making a package for a package
-        management system like RPM, you have to specify for each package what
-        its dependencies are, but there are no guarantees that this
-        specification is complete. If you forget a dependency, then the
-        component will build and work correctly on <em>your</em> machine if
-        you have the dependency installed, but not on the end user's machine
-        if it's not there.</p>
-
-      <p>Since Nix on the other hand doesn’t install packages in “global”
-        locations like <tt>/usr/bin</tt> but in package-specific directories,
-        the risk of incomplete dependencies is greatly reduced. This is
-        because tools such as compilers don’t search in per-packages
-        directories such as
-        <tt>/nix/store/5lbfaxb722zp…-openssl-0.9.8d/include</tt>, so if a
-        package builds correctly on your system, this is because you specified
-        the dependency explicitly.</p>
-
-      <p>Runtime dependencies are found by scanning binaries for the hash
-        parts of Nix store paths (such as <tt>r8vvq9kq…</tt>). This may sound
-        risky, but it works extremely well.</p>
+        complete.</p>
+      
+      <p>When you’re making a package for a package
+        management system like RPM, you are supposed to declare its dependencies,
+        but you can't easily guarantee that your dependency declaration is complete.
+        If you forget a dependency, then the
+        component may build and work correctly on <em>your</em> machine, if
+        you already have the dependency installed, while failing on the end user's
+        machine, if they don't.</p>
+
+      <p>Under Nix, a build process will only find resources that have been
+        declared explicitly as dependencies. There's no way it can build until
+        everything it needs has been correctly declared. If it builds, you will know
+        you've provided a complete declaration.</p>
+
+      <p>Once a build is complete, ongoing runtime dependencies are detected
+        automatically by scanning the package's build result files for the hash parts
+        of Nix store paths (such as <tt>r8vvq9kq…</tt>). This may sound risky,
```

Comment:
Maybe mention that it is enough to find at least one copy, not all of them? This adds confidence, and judging from some other discussions, it is a point worth an emphasis to give people a clearer picture.

Reply:
Features page shouldn't have to explain how scanning works just that runtime dependencies are detected.

Reply:
I don't see how that would be clarifying. Which other discussions? Perhaps there will turn out to be a better way of saying whatever it was they found.

Implying/letting on that it might not find all of them doesn't seem like it would instill confidence, to me.